lib/attrsets.nix

```diff
@@ -43,6 +44,16 @@ rec {
       [ { name = head attrPath; value = setAttrByPath (tail attrPath) value; } ];
 
 
+  /* Map function f over the value of the attribute described by attrPath.
+     For instance, mapAttrByPath [ "x" "y" ] (a: a+1) { x.y = 1; } returns
+     { x.y = 2; }. If the path doesn’t exist the default value is used. */
```

Comment:
It's not clear to be from this description what is being mapped. That suggests that multiple values are updated, but the example suggests only the specified attribute (x.y) is updated. And what is the `recursiveUpdate` for?